installer/engine_installer/engine_installer.go

```diff
@@ -14,8 +14,168 @@
+type EngineInstaller struct {
+	BridgeNetwork   string `json:"bridge-net"`
+	PublicNetwork   string `json:"public-net"`
+	ImageStore      string `json:"img-store"`
+	ComputeResource string `json:"compute"`
+	Target          string `json:"target"`
+	User            string `json:"user"`
+	Password        string `json:"password"`
+	Name            string `json:"name"`
+	Thumbprint      string `json:"thumbprint"`
+	CreateCommand   string
+}
+
+// AuthHTML holds the invalid login variable
+type AuthHTML struct {
+	InvalidLogin bool
+}
+
+// ExecHTMLOptions contains fields for html templating in exec.html
+type ExecHTMLOptions struct {
+	BridgeNetwork   template.HTML
+	PublicNetwork   template.HTML
+	ImageStore      template.HTML
+	ComputeResource template.HTML
+	Target          string
+	User            string
+	Password        string
+	Name            string
+	Thumbprint      string
+	CreateCommand   string
+}
+
+var (
+	validator *validate.Validator
+)
+
+// NewEngineInstaller returns a new EngineInstaller struct with empty parameters
+func NewEngineInstaller() *EngineInstaller {
+	return &EngineInstaller{Name: "default-vch"}
+}
+
+func (ei *EngineInstaller) populateConfigOptions() *EngineInstallerConfigOptions {
+	vc := validator.IsVC()
+	fmt.Printf("Is VC: %t\n", vc)
+
+	dcs, err := validator.ListDatacenters()
+	if err != nil {
+		fmt.Println(err)
+		return nil
+	}
+	for _, d := range dcs {
+		fmt.Printf("DC: %s\n", d)
+	}
+
+	comp, err := validator.ListComputeResource()
+	if err != nil {
+		fmt.Println(err)
+		return nil
+	}
+	for _, c := range comp {
+		fmt.Printf("compute: %s\n", c)
+	}
+
+	rp, err := validator.ListResourcePool("*")
+	if err != nil {
+		fmt.Println(err)
+		return nil
+	}
+	for _, p := range rp {
+		fmt.Printf("rp: %s\n", p)
+	}
+
+	nets, err := validator.ListNetworks(!vc) // set to false for vC
+	if err != nil {
+		fmt.Println(err)
+		return nil
+	}
+	for _, n := range nets {
+		fmt.Printf("net: %s\n", n)
+	}
+
+	dss, err := validator.ListDatastores()
+	if err != nil {
+		fmt.Println(err)
+		return nil
+	}
+	for _, d := range dss {
+		fmt.Printf("ds: %s\n", d)
+	}
+
+	return &EngineInstallerConfigOptions{
+		Networks:      nets,
+		Datastores:    dss,
+		ResourcePools: rp,
+	}
+}
+
+func (ei *EngineInstaller) buildCreateCommand(binaryPath string) {
+	var createCommand []string
+
+	createCommand = append(createCommand, binaryPath+"/vic/vic-machine-linux")
+	createCommand = append(createCommand, "create")
+	createCommand = append(createCommand, "--no-tlsverify")
+	createCommand = append(createCommand, fmt.Sprintf("--target %s", ei.Target))
+	createCommand = append(createCommand, fmt.Sprintf("--user %s", ei.User))
+	createCommand = append(createCommand, fmt.Sprintf("--password %s", ei.Password))
+	createCommand = append(createCommand, fmt.Sprintf("--name %s", ei.Name))
+	createCommand = append(createCommand, fmt.Sprintf("--public-network %s", ei.PublicNetwork))
```

Comment:
Hmm... I've made a change here. The args in the exec.Command args slice get directly passed argv for execution, so it should be fine.  

ref:https://groups.google.com/forum/#!topic/golang-nuts/qYVlLVR5504